add List-form ApplicationUser

BannikovRuslan opened the pull request.

BannikovRuslan committed:
feat: добавлен интерфейс меню лист-формы

BannikovRuslan committed:
fix: code smells and other

AntonIV8 commented on src/AndroidODataOfflineSample/app/src/main/java/com/flexberry/androidodataofflinesample/ui/ApplicationUserListFormActivity.kt:
```diff
@@ -0,0 +1,32 @@
+import com.flexberry.androidodataofflinesample.ui.theme.AndroidODataOfflineSampleTheme
+
+class ApplicationUserListFormActivity : ComponentActivity() {
+    @RequiresApi(Build.VERSION_CODES.O)
```
А вот это зачем ? Вроде в MainActivity такого не создавалось

AntonIV8 commented on src/AndroidODataOfflineSample/app/src/main/java/com/flexberry/androidodataofflinesample/ui/applicationuserlistformmodel/ApplicationUserListFormViewModel.kt:
```diff
@@ -0,0 +1,25 @@
+class ApplicationUserListFormViewModel {
+    data class MainViewModelState (val isOnline: Boolean)
+
+    data class User(
```
Тут как-то надо использовать модель 
com.flexberry.androidodataofflinesample.data.model.ApplicationUser

В MainViewModel там сейчас state (если ты по аналогии делал). State это другое. По ним отдельная еще задача будет

AntonIV8 commented on src/AndroidODataOfflineSample/app/src/main/java/com/flexberry/androidodataofflinesample/ui/applicationuserlistformmodel/SampleData.kt:
```diff
@@ -0,0 +1,34 @@
+object SampleData {
+    @RequiresApi(Build.VERSION_CODES.O)
+    val usersSample = listOf(
+        User(
```
тут тоже должны быть

com.flexberry.androidodataofflinesample.data.model.ApplicationUser

AntonIV8 commented on src/AndroidODataOfflineSample/app/src/main/java/com/flexberry/androidodataofflinesample/ui/applicationuserlistformmodel/ApplicationUserListFormViewModel.kt:
```diff
@@ -0,0 +1,25 @@
+import java.time.LocalDate
+
+class ApplicationUserListFormViewModel {
+    data class MainViewModelState (val isOnline: Boolean)
```
А это зачем ?

BannikovRuslan committed:
fix: в качестве тестовых данных используется список из объектов класса ApplicationUser +  правки по интерфейсу главного меню и др.

VChurekov commented on src/AndroidODataOfflineSample/app/src/main/java/com/flexberry/androidodataofflinesample/ui/applicationuserlistformmodel/SampleData.kt:
```diff
@@ -0,0 +1,284 @@
+import java.sql.Timestamp
+import java.util.UUID
+
+object SampleData {
```
Добавь сюда @Suppress("kotlin:S1192"). Чтобы code smells не ругался.

BannikovRuslan committed:
fix: code smells - удалены дубликаты данных

BannikovRuslan committed:
fix: добавлен @Suppress("kotlin:S1192")

BannikovRuslan committed:
fix: DuplicatedBlocks

BannikovRuslan committed:
fix:

BannikovRuslan committed:
fix:

BannikovRuslan committed:
fix

BannikovRuslan committed:
fix

BannikovRuslan committed:
fix*

BannikovRuslan committed:
fix: удален SampleData

BannikovRuslan committed:
Merge remote-tracking branch 'origin/develop' into feature-232223--add-list-form-applicationuser

BannikovRuslan committed:
fix: удалено лишнее
